engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobUpdateTimeoutTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import io.camunda.zeebe.scheduler.clock.ActorClock;
+import io.camunda.zeebe.test.util.Strings;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class JobUpdateTimeoutTest {
```

Comment:
❌ I thought of another test case I'd like to see, which is updating the timeout multiple times:

1. create a job
2. Activate the job
3. Update the timeout
4. Update the timeout again
5. Assert that the deadline is the current time + the timeout of step 4

I believe this test is relevant because this way we verify that we are actually updating the deadline in the Job ColumnFamily, as well as the deadlines ColumnFamily. If we wouldn't the second update timeout would be rejected.

Reply:
I want 2 more test cases in the `JobTimeOutTest`. This test class verifies we are timing out jobs correctly.

**`shouldTimeOutAfterDecreasingTimeout`**

1. Create a job
2. Activate job with a high timeout
3. Update job to a lower timeout
4. Increase the time
5. Verify the job has expired

**`shouldTimeOutAfterIncreasingTimeout`**

1. Create a job
2. Activate job with a high timeout
3. Update job to an even higher timeout
4. Increase the time
5. Verify the job has expired

Reply:
Done 👍